src/cudadecoder/batched-threaded-nnet3-cuda-online-pipeline.cc

```diff
@@ -31,6 +33,11 @@
 
 namespace kaldi {
 namespace cuda_decoder {
+
+const double kSleepForCallBack = 10e-3;
```

Comment:
Probably compiles to the same thing, but couldn't we use constexpr here in theory?

Reply:
Yes and yes. :) I haven't seen a single compiler treating const qualification and constexpr declaration of simple numeric types differently: before constexpr, const was a de-facto guarantee of the literal constant substitution, and undoing that, although legal per C++11, would affect a lot of existing code. But constexpr indeed provides the compile-time evaluation guarantee.

I ultimately want to use [the `chrono` literals](https://en.cppreference.com/w/cpp/chrono/operator%22%22us) for time constants anyway, like `Sleep(250ms)` or `constexpr auto kDelay = 250ms;`. They are too sweet and readable not to use them. :) Do you mind if we leave them as is in the meantime?

Reply:
I agree that literals would be perfect - but for now it's fine. I'll update some other parts of the code to make it more C++14-like too (like finally being able to use std::make_unique :) )

Reply:
Oh yeah, the delightsome long coveted `std::make_unique` may at last be used!

I once heard someone referring to C++14 as "C++11 service pack 1".

And since we're now allowed to use C++14, we can finally begin craving the C++17 features! :-D